File: src/modifiers/minimumScaleFactor.ts
import { createModifier } from './createModifier';

export interface MinimumScaleFactorParams {
  /// Smallest fraction of the font size text may shrink to (0...1) before
  /// truncating instead. E.g. `0.5` allows shrinking to half size.
  value?: number;
}

/// SwiftUI `.minimumScaleFactor(_:)`. Lets text within the view shrink to
/// fit available space down to the given fraction of its font size. Pass
/// a number directly or an object.
export function minimumScaleFactor(params?: MinimumScaleFactorParams | number) {
  if (typeof params === 'number') {
    return createModifier('minimumScaleFactor', { value: params });
  }
  return createModifier('minimumScaleFactor', params);
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/MinimumScaleFactor.swift
import SwiftUI

/// SwiftUI `.minimumScaleFactor(_:)`. Lets text within the view shrink to
/// fit available space down to the given fraction of its font size. An
/// absent `value` leaves the view unchanged.
enum RNWMinimumScaleFactorModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("minimumScaleFactor") { view, params, _ in
            guard let factor = params.double("value") else { return view }
            return AnyView(view.minimumScaleFactor(factor))
        }
    }
}
